### ISSUE TITLE
BanditBear: Tests to execute option missing

### ISSUE THREAD

**Issue by sangamcse** [coala/coala-bears#2386]
`bandit` has an option `-t comma-separated list of test IDs to run` which is missing in BanditBear.

difficulty/low

**Comment by sangamcse** [comment]
Please assign this to me 😃

**Comment by jayvdb** [comment]
Why is this option useful?

**Comment by sangamcse** [comment]
Sometime we wants only a few tests to run. We can do that in our current setting of BanditBear by giving a long list of tests to skip in `bandit_skipped_tests` or we can add an option(in this issue) to run some specific tests. 

Note: I found it odd that we don't have an option to run some specific tests. There is a lot of option https://github.com/openstack/bandit#usage

**Comment by sangamcse** [comment]
For example: what if I wants to run only `B112  try_except_continue` test? Then I have to give a long list of tests ID to `bandit_skipped_tests` list and that is not a good option.

**Comment by jayvdb** [comment]
ok, we need to look at other bears which have similar lists of checks, and work towards a consistent settings system for these.  pycodestyle and pydocstyle do have similar lists of checks and command line options to include/select or ignore various checks.

**Comment by jayvdb** [comment]
@sangamcse , we dont care what the linter API is.  coala wraps linters to *hide* their differences.  We provide setting names which are more consistent, to make it easier for users to manage many different tools.

**Comment by sangamcse** [comment]
@jayvdb, So I looked at other bears which have similar lists of checks and I have noted their command type also. Here it is:
GoErrCheckBear --> `-ignore ignore_list` and `-ignorepkg ignorepkg_list`
ArtisticStyleBear --> bool type list of checks
CPPCheckBear --> `--enable list_of_issues_type`
CPPLintBear --> `--filter=ignore_list,include_list`
HTMLLintBear --> `--disable=htmllint_ignore_list`
HTTPoliceLintBear --> `-s=httpolice_silence_ids_list`
PycodestyleBear --> `--ignore=ignore_checks_list` and `--select=select_checks_list`
PydocstyleBear --> similar to PycodestyleBear
PyLintBear --> `--disable=pylint_disable_ids_list` and `--enable=pylint_enable_ids_list`
RSTcheckBear --> `--ignore-language=code_block_language_ignore_list`, `--ignore-directives=directive_ignore_list` and `--ignore-roles=role_ignore_list`
ShellCheckBear --> `-e list_of_ignore_rules`

Note: I have tried to write a [CheckSelector](https://github.com/coala/coala-bears/pull/2390/files#diff-565b15f3772ba98b4bcaa248fd442350R1). Please suggest me and correct me if I'm doing anything wrong. 😃

**Comment by jayvdb** [comment]
Note there is also https://github.com/coala/coala-bears/pull/2257 , by @manankalra

**Comment by jayvdb** [comment]
> RSTcheckBear --> --ignore-language=code_block_language_ignore_list, --ignore-directives=directive_ignore_list and --ignore-roles=role_ignore_list

I think these are a slightly different concept, especially directive & roles, as these are ignoring syntax in the rst which rstcheck doesnt know about.

**Comment by sangamcse** [comment]
Oh. I tried to find this PR based on your suggestion earlier on gitter but was unable to find it.
So, what I wrote in ChecksSelector can work with RSTcheckBear. But real difficulty is with bool type.

**Comment by jayvdb** [comment]
Right. So the remaining design question is whether we should be trying to merge the two varieties:

1. lots of bool args with (sometimes descriptive) names, where the arg names are hard-coded per linter release, and the bear needs updating for each new linter release
2. select/unselect args with opaque codes, with codes added in each release but the bear doesnt need modification

1. is beyond the scope of this BanditBear bug, but clearly it is the bigger maintenance/usability problem for us, and it is the true abstraction of our settings from the linters cmdline.  Ideally we solve both at the same time, but that will definitely require a few enhancements to coalib.  Lets defer that for the moment.

What is in the scope of this issue is select/unselect are two mutually exclusive lists, and at least that is a coalib core enhancement to `Settings`.  But you have a decent working version which can get merged into BanditBear before being migrated to coalib.

Also in the scope of this issue is the settings names should be rather consistent, and have prefixes for each bear, so that the user can have

```
bears = FooBear, BarBear
foo_select = a, b, c
bar_ignore = e, f, g
```

We need to determine the best naming convention, based on existing bears if possible.
Then after this, we can update the other bears to comply with the new naming convention.
